packages/pds/src/api/com/atproto/server/getAccountInviteCodes.ts

```diff
@@ -11,72 +11,39 @@ export default function (server: Server, ctx: AppContext) {
       const requester = auth.credentials.did
       const { includeUsed, createAvailable } = params
 
-      const accntSrvc = ctx.services.account(ctx.db)
-
-      const [user, userCodes] = await Promise.all([
-        ctx.db.db
-          .selectFrom('account')
-          .where('did', '=', requester)
-          .select(['invitesDisabled', 'createdAt'])
-          .executeTakeFirstOrThrow(),
-        accntSrvc.getAccountInviteCodes(requester),
+      const [account, userCodes] = await Promise.all([
+        ctx.accountManager.getAccount(requester),
+        ctx.accountManager.getAccountInvitesCodes(requester),
       ])
+      if (!account) {
+        throw new InvalidRequestError('Account not found', 'NotFound')
+      }
 
-      let created: string[] = []
+      let created: CodeDetail[] = []
 
-      const now = new Date().toISOString()
       if (
         createAvailable &&
         ctx.cfg.invites.required &&
         ctx.cfg.invites.interval !== null
       ) {
-        const { toCreate, total } = await calculateCodesToCreate({
+        const { toCreate } = calculateCodesToCreate({
           did: requester,
-          userCreatedAt: new Date(user.createdAt).getTime(),
+          userCreatedAt: new Date(account.createdAt).getTime(),
           codes: userCodes,
           epoch: ctx.cfg.invites.epoch,
           interval: ctx.cfg.invites.interval,
         })
         if (toCreate > 0) {
-          created = genInvCodes(ctx.cfg, toCreate)
-          const rows = created.map((code) => ({
-            code: code,
-            availableUses: 1,
-            disabled: user.invitesDisabled,
-            forAccount: requester,
-            createdBy: requester,
-            createdAt: now,
-          }))
-          await ctx.db.transaction(async (dbTxn) => {
-            await dbTxn.db.insertInto('invite_code').values(rows).execute()
-            const finalRoutineInviteCodes = await dbTxn.db
-              .selectFrom('invite_code')
-              .where('forAccount', '=', requester)
-              .where('createdBy', '!=', 'admin') // dont count admin-gifted codes aginast the user
-              .selectAll()
-              .execute()
-            if (finalRoutineInviteCodes.length > total) {
-              throw new InvalidRequestError(
-                'attempted to create additional codes in another request',
-                'DuplicateCreate',
-              )
-            }
```

Comment:
I noticed this check disappeared (i.e. does not occur in `createAccountInviteCodes()`), just ensuring it's intentional.

Reply:
this is a similar thing to racing creates with an invite code

basically, i'm not sure about doing the integrity check _within_ a sqlite tx. but maybe i'm being over-paranoid 🤔

Reply:
kk added back in the race prevention here as well 👌